### DISCUSSION TITLE
Disable automatic replacement of certain math symbols (==, !=, <, <=>, etc.)

### DISCUSSION THREAD

**Discussion by falbrechtskirchinger** [mkdocs/mkdocs#2820] [Q&A] [answered]
Hello,

I'm writing some documentation for a project and noticed that mkdocs automatically replaces certain math symbols in code sections. For example, \`<=>\` becomes ![Figure](https://latex.codecogs.com/png.image?\dpi{110}&space;\bg_white&space;\iff) (aka. iff).
I've tried commenting out `markdown_extensions:` and even `extra_javascript:` (which loads MathJax) to no avail.
How can I disable these replacements either globally or inline?

Here's the project's `mkdocs.yml`:
https://github.com/nlohmann/json/blob/develop/doc/mkdocs/mkdocs.yml

Thanks!

**Comment by Andre601** [comment]
Have you tried to just escaping them using a backslash? For example: `\<=>`

**Reply by falbrechtskirchinger** [reply]
I have without success.

**Reply by facelessuser** [reply]
This is almost certainly done by either a plugin, or some ligature translations in the web browser, probably enabled by some CSS styling.

I see SmartSymbols was commented out, you can turn off individual symbols via the config, but since you've turned it off completely, there is no way it is translating any at least.

**Reply by falbrechtskirchinger** [reply]
> This is almost certainly done by either a plugin, or some ligature translations in the web browser, probably enabled by some CSS styling.

If this a browser thing it's done by both Chrome and Firefox without any extensions/plugins or non-default settings (confirmed with fresh profiles).
I also don't see any custom CSS in the `mkdocs.yml` sub directories. If it is a mkdocs plugin or default CSS the question stands: how do I turn it off?

> I see SmartSymbols was commented out, you can turn off individual symbols via the config, but since you've turned it off completely, there is no way it is translating any at least.

I did check the docs on Markdown extensions and didn't see any mention of the particular symbols being transformed.

I have also explicitly enabled asciimath in MathJax and the iff symbol looks slightly different (shorter, stubbier), than the one I'm seeing. (And like I said in the OP I comented out MathJax and it made no difference.)

Any other ideas?

**Reply by facelessuser** [reply]
Yes, both Firefox and Chrome support ligatures. It is most certainly that you are using a font that supports ligatures and it is translating. This is not a MkDocs issue or Python Markdown issue as neither convert symbols in this way by default. I see that the project you linked to is using Material theme, so it isn't even a problem with the MkDocs default themes.

I looked at your rendered code blocks, and I see the that it is indeed using ligatures. Try adding this CSS.

```css
code, pre {
    font-variant-ligatures: none;
}
```

**Reply by falbrechtskirchinger** [reply]
Thanks! That did it. I was completely unaware that's even a thing in CSS.

If you could re-post your reply as a separate answer, I'll mark it as such.

**Reply by facelessuser** [reply]
It appears you are using JetBrains Mono for the code font. That font is using ligatures.

```
  font:
    text: Roboto
    code: JetBrains Mono
```

**Reply by facelessuser** [reply]
It doesn't 

> Yes, both Firefox and Chrome support ligatures. It is most certainly that you are using a font that supports ligatures and it is translating. This is not a MkDocs issue or Python Markdown issue as neither convert symbols in this way by default. I see that the project you linked to is using Material theme, so it isn't even a problem with the MkDocs default themes.
> 
> I looked at your rendered code blocks, and I see the that it is indeed using ligatures. Try adding this CSS.
> 
> ```css
> code, pre {
>     font-variant-ligatures: none;
> }
> ```
>
> It appears you are using JetBrains Mono for the code font. That font is using ligatures.
> 
> ```
>   font:
>     text: Roboto
>     code: JetBrains Mono
> ```

You could also use a font that doesn't use ligatures 🙂

**Reply by facelessuser** [reply]
Sorry , it didn't post it separate for some reason 🤷🏻

**Reply by falbrechtskirchinger** [reply]
There're two text boxes one with the placeholder text "Write a reply" and another one below with the placeholder text "Suggest an answer". Easy to miss. Use the second one.

**Reply by falbrechtskirchinger** [reply]
> You could also use a font that doesn't use ligatures :slightly_smiling_face:

I'm a mere contributor. I'm not in charge of these decisions. ;-)

**Reply by facelessuser** [reply]
Fair enough 🙂

**Comment by facelessuser** [comment, marked as answer]
> Yes, both Firefox and Chrome support ligatures. It is most certainly that you are using a font that supports ligatures and it is translating. This is not a MkDocs issue or Python Markdown issue as neither convert symbols in this way by default. I see that the project you linked to is using Material theme, so it isn't even a problem with the MkDocs default themes.
> 
> I looked at your rendered code blocks, and I see the that it is indeed using ligatures. Try adding this CSS.
> 
> ```css
> code, pre {
>     font-variant-ligatures: none;
> }
> ```
>
> It appears you are using JetBrains Mono for the code font. That font is using ligatures.
> 
> ```
>   font:
>     text: Roboto
>     code: JetBrains Mono
> ```

You could also use a font that doesn't use ligatures 🙂

**Reply by falbrechtskirchinger** [reply]
That worked. Thanks again!
